From 96bfc02133ace26d63d1c85fbfc3a686262f6607 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Wed, 9 Nov 2011 10:18:58 +0100
Subject: [PATCH] 50plymouth/plymouth-pretrigger.sh: respect primary console
 setting

The kernel's primary console device is determined by the last "console="
argument on the kernel command line. This setting should be respected by
dracut-generated initial RAM disks.

Steps to Reproduce:
(Easiest using a KVM VM, virt-manager and "virsh console")
1. Boot with a kernel command line ending in
 console=tty0 console=ttyS0,115200
2. Observe both tty0 and ttyS0.

The output of init scripts is sent to ttyS0, as the final "console="
argument determines the primary console device as per
Documentation/serial-console.txt in the kernel sources.

https://bugzilla.redhat.com/show_bug.cgi?id=752073
---
 modules.d/50plymouth/plymouth-pretrigger.sh | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/modules.d/50plymouth/plymouth-pretrigger.sh b/modules.d/50plymouth/plymouth-pretrigger.sh
index 725d7e7..4a3fcdf 100755
--- a/modules.d/50plymouth/plymouth-pretrigger.sh
+++ b/modules.d/50plymouth/plymouth-pretrigger.sh
@@ -16,8 +16,11 @@ if ! getarg rd_NO_PLYMOUTH; then
     [ -c /dev/hvc0 ] || mknod -m 0600 /dev/hvc0 c 229 0
 
     info "Starting plymouth daemon"
+    consoledev=$(getarg console= | sed -e 's/,.*//')
+    consoledev=${consoledev:-tty0}
+    [ -x /lib/udev/console_init ] && /lib/udev/console_init "/dev/$consoledev"
+ 
     [ -x /bin/plymouthd ] && /bin/plymouthd --attach-to-session
-    /lib/udev/console_init tty0
     /bin/plymouth --show-splash 2>&1 | vinfo
 fi
 
